### ISSUE TITLE
Example in example.py using resources

### ISSUE THREAD

**Issue by adamliter** [presleyp/Speriment#52]
I'm wondering if it would be possible to add an example to `example.py` that uses resources. I'm currently trying to design an experiment with Speriment where there are sound files that should play on each page, but I'm not able to get this to work.

In `example.py`, I have:

```python
with make_experiment(IDGenerator()):
    training_stimuli = get_dicts('TrainingStimuli_Exp4_List1.csv')
    training_pages = [Page(
                           [instruction, row['STIMULUS']],
                           resources = row['SOUND'],
                           tags = {'PageType': 'Training'}
                           ) for row in training_stimuli]
    training_block = Block(pages = training_pages)
    experiment = Experiment([training_block])
    experiment.install('example_experiment')
```

And `TrainingStimuli_Exp4_List1.csv` looks like this:

```
STIMULUS,SOUND
pupa,Recordings/pupa.wav
vizu,Recordings/vizu.wav
pupi,Recordings/pupi.wav
zava,Recordings/zava.wav
ziva,Recordings/ziva.wav
bubi,Recordings/bubi.wav
sasu,Recordings/sasu.wav
vuzi,Recordings/vuzi.wav
sifi,Recordings/sifi.wav
putu,Recordings/putu.wav
```

The sound files are inside of a folder called `Recordings`, which is in the top-level directory for the project. This isn't working. I've also tried several variations of this: putting `Recordings` inside of `static`, putting the `.wav` files inside of the top-level directory for the project without any enclosing folder, changing the `.csv` file to use `./Recordings/` instead of `Recordings/`, etc.. None of these variations (with the relevant changes in the `.csv` files) has worked.

I assume I'm somehow specifying the name of the file incorrectly, so it would be useful if you could add an example that uses some sort of resource (image, audio, and/or video) to `example.py`.

Thanks for a great tool, by the way! :smile_cat:

**Comment by presleyp** [comment]
Great question! I think your first way is correct and it just needs to be in a list, like `resources =['Recordings/pupa.wav']`. So the general rule is: put resources anywhere in the project directory and refer to them relative to the top level of the project directory.

I'm working on a new version with improved documentation, I'll make sure to cover this there.

**Comment by adamliter** [comment]
Changing `resources = row['SOUND'],` to `resources = [row['SOUND']],` does seem to bring things closer to working, but things don't quite yet work as I'm hoping.

An audio player shows up in both Safari and Chrome. In Firefox, the audio player flashes on the screen, but then it disappears immediately after the page loads.

In none of these three browsers does the audio actually play, however. In Safari and Chrome, it looks like it's trying (and perhaps failing?) to load the audio file. In Firefox, I'm not entirely sure what's going on since the audio player just disappears.

Ideally, I'd like the audio to just play once when the page is loaded; I don't necessarily want an audio player to actually be visible on the screen.

This does show up in the console, so perhaps this is part of the problem? Are these files that are supposed to be provided by PsiTurk?

<img width="1025" alt="screen shot 2016-02-09 at 11 55 58 am" src="https://cloud.githubusercontent.com/assets/4974404/12923539/1c2e9180-cf24-11e5-9af3-ead80e6cbf80.png">

Thanks for the quick response and the suggestions! I really appreciate it. Great to hear about the improved documentation and new version! :smile:

**Comment by adamliter** [comment]
That previous image was from the console in Safari. Google Chrome's console also shows this:

<img width="625" alt="screen shot 2016-02-09 at 1 17 03 pm" src="https://cloud.githubusercontent.com/assets/4974404/12926063/879b55ce-cf2f-11e5-9a90-527fa0afb9e9.png">

This is despite the fact that the Recordings folder is in the top-level directory for the site, and all of the `.wav` files are there inside the Recordings folder. Any idea what might be going on?

Thanks again for your help and responses!

**Comment by presleyp** [comment]
Oh, the map files...I've had that problem before and I can't remember how I solved it. I'm not sure if it actually matters though. Might try an `npm install` if you haven't already. Unfortunately I'm not sure what's going on with the resources though! I'll take a look this week. I can tell you that I don't have it set up to just play once; I had decided to do the players because you never know what's going on for the participant - and ambulance drives by and they miss the sound or something. But if it's important for your design, go ahead and make an issue for it.

**Comment by adamliter** [comment]
You're right. According to some discussion on [this Stack Overflow question](http://stackoverflow.com/q/18365315/2571049), it looks like the map files or just for debugging purposes, so they aren't necessarily needed.

I'm not sure what's going on with the 500 internal server error, though, when it comes to loading the sound files. I tried checking the permissions for the folder and sound files, but that doesn't seem to be the problem. They're set to `-rwxr-xr-x`, so they should be readable and executable by anybody.

If you have any thoughts, I'd greatly appreciate it!

**Comment by adamliter** [comment]
I had some help in figuring out what the problem was (shout out to @brettimus). It seems that Flask (which is something that I think PsiTurk—not Speriment—relies on, correct?) doesn't like serving static resources from anywhere other than the `static` folder.

Looking at the server log showed something like this:

```
2016-02-09 18:05:56,775 Exception on /Recordings/vizu.wav [GET]
Traceback (most recent call last):
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/psiturk/experiment.py", line 667, in regularpage
    return render_template(foldername+"/"+pagename)
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/flask/templating.py", line 127, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/jinja2/environment.py", line 851, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/jinja2/environment.py", line 812, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/jinja2/environment.py", line 774, in _load_template
    cache_key = self.loader.get_source(self, name)[1]
  File "/Users/adamliter/.virtualenvs/speriment-test/lib/python2.7/site-packages/flask/templating.py", line 64, in get_source
    raise TemplateNotFound(template)
TemplateNotFound: Recordings/vizu.wav
```

However, putting the `Recordings` folder inside of static and then changing the `.csv` file to the following worked:

```
STIMULUS,SOUND
pupa,/static/Recordings/pupa.wav
vizu,/static/Recordings/vizu.wav
pupi,/static/Recordings/pupi.wav
zava,/static/Recordings/zava.wav
ziva,/static/Recordings/ziva.wav
bubi,/static/Recordings/bubi.wav
sasu,/static/Recordings/sasu.wav
vuzi,/static/Recordings/vuzi.wav
sifi,/static/Recordings/sifi.wav
putu,/static/Recordings/putu.wav
```

So adding something to the documentation about needing to house static assets inside of the `static` folder would definitely be very helpful.

Thanks again, Presley! :smile:

**Comment by presleyp** [comment]
Oh, good to know! I had put my resources in `static` so I thought once I figured out how to refer to them that was it.

**Comment by brettimus** [comment]
:tada: Yay I got internet points today!
